typescript/cli/src/context/strategies/chain/chainResolver.ts

```diff
@@ -0,0 +1,279 @@
+import {
+  ChainName,
+  DeployedCoreAddresses,
+  DeployedCoreAddressesSchema,
+  EvmCoreModule,
+} from '@hyperlane-xyz/sdk';
+import { ProtocolType, assert } from '@hyperlane-xyz/utils';
+
+import { CommandType } from '../../../commands/signCommands.js';
+import { readCoreDeployConfigs } from '../../../config/core.js';
+import { getWarpRouteDeployConfig } from '../../../config/warp.js';
+import { RebalancerConfig } from '../../../rebalancer/config/RebalancerConfig.js';
+import {
+  runMultiChainSelectionStep,
+  runSingleChainSelectionStep,
+} from '../../../utils/chains.js';
+import {
+  getWarpConfigs,
+  getWarpCoreConfigOrExit,
+} from '../../../utils/warp.js';
+import { requestAndSaveApiKeys } from '../../context.js';
+
+/**
+ * Resolves chains based on command type.
+ * @param argv - Command line arguments.
+ * @returns Promise<ChainName[]> - The chains resolved based on the command type.
+ */
+export async function resolveChains(
+  argv: Record<string, any>,
+): Promise<ChainName[]> {
+  const commandKey = `${argv._[0]}:${argv._[1] || ''}`.trim() as CommandType;
+
+  switch (commandKey) {
+    case CommandType.WARP_DEPLOY:
+      return resolveWarpRouteConfigChains(argv);
+    case CommandType.WARP_SEND:
+    case CommandType.SEND_MESSAGE:
+    case CommandType.STATUS:
```

Comment:
does this mean `hyperlane status` will no longer ask me for PKs unless i add `--relay`? and specifically it will only ask me for PKs relevant to the message being relayed?